qiskit_experiments/library/calibration/rough_amplitude_cal.py

```diff
@@ -182,7 +182,8 @@ def update_calibrations(self, experiment_data: ExperimentData):
 
         for angle, param, schedule, prev_amp in experiment_data.metadata["angles_schedules"]:
 
-            value = np.round(angle / rate, decimals=8) * np.exp(1.0j * np.angle(prev_amp))
+            # This implementation conserves the type, while working for both real and complex prev_amp
+            value = np.round(angle / rate, decimals=8) * (prev_amp / np.abs(prev_amp))
```

Comment:
```suggestion
            value = np.round(angle / rate, decimals=8) * prev_amp / np.abs(prev_amp)
```
are the brackets needed here? If not I'd remove